src/System.Buffers/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace System.Buffers
+{
+    [EventSource(
+        Guid = "20b30044-b729-457e-8dda-8b41b5dd02e6", 
+        Name = "System.Buffers.DefaultArrayPoolEvents",
+        LocalizationResources = "FxResources.System.Buffers.SR")]
+    internal sealed class ArrayPoolEventSource : EventSource
+    {
+        internal readonly static ArrayPoolEventSource Log = new ArrayPoolEventSource();
+
+        [Event(1, Level = EventLevel.Informational, Opcode = EventOpcode.Start)]
+        internal void BufferRented(int bufferId, int bufferSize, int bucketId) { WriteEvent(1, bufferId, bufferSize, bucketId); }
+
+        [Event(2, Level = EventLevel.Informational, Opcode = EventOpcode.Stop)]
+        internal void BufferReturned(int bufferId) { WriteEvent(2, bufferId); }
+
```

Comment:
I think rented/returned are informational. But should the ones logged when an allocation happens be warnings?